### DISCUSSION TITLE
How to obtain the coordinate information of the cursor during text input or deletion?

### DISCUSSION THREAD

**Discussion by kimmy-wang** [ether/etherpad-lite#5123] [Q&A]
1. How to obtain the coordinate information of the cursor during text input or deletion?
2. When deleting text, how to get the deleted text?

**Comment by webzwo0i** [comment]
From the `rep` object that contains `selStart`/`selEnd` of type [row, col].

E.g. you can hook the `aceKeyEvent` that is called when typing/deleting etc.

**Reply by kimmy-wang** [reply]
I have a requirement, which is to implement a method to achieve the effect of manually inputting text by calling this method. Do you have any comments or suggestions for implementing this method?

**Reply by rhansen** [reply]
Just manipulate the DOM how you want (e.g., by inserting text nodes).

**Reply by kimmy-wang** [reply]
I noticed that when entering or deleting text, a message event is triggered.
![screenshot-20210723-090738](https://user-images.githubusercontent.com/19725091/126726902-3cacedf8-c87b-4442-97f7-a09224c49850.png)

**Reply by webzwo0i** [reply]
Please take a look at the various documentation available. The screenshot refers to a changeset, ie whenever you edit the document it calculates a normalized representation of the change and transmits it to the server. This functionality is the base of the sync protocol Etherpad is built around. Whenever you manipulate the DOM or pad, such a changeset is automatically calculated and transmitted - this is intended behavior.

However, we also provide a lot of different hooks that you can use to work with the editor.
